os_migrate/plugins/module_utils/resource.py

```diff
@@ -158,15 +158,22 @@ def create_or_update(self, conn):
         """
         refs = self._refs_from_ser(conn)
         sdk_params = self._to_sdk_params(refs)
-        existing = self._find_sdk_res(conn, sdk_params['name'])
-        if existing:
-            if self._needs_update(self.from_sdk(conn, existing)):
-                self._update_sdk_res(conn, sdk_params['name'], sdk_params)
+
+        # Not every resource has the 'name' key
+        if 'name' in sdk_params:
+            existing = self._find_sdk_res(conn, sdk_params['name'])
+            if existing:
+                if self._needs_update(self.from_sdk(conn, existing)):
+                    self._update_sdk_res(conn, sdk_params['name'], sdk_params)
+                    return True
+            else:
+                self._create_sdk_res(conn, sdk_params)
                 return True
+            return False  # no change done
         else:
-            self._create_sdk_res(conn, sdk_params)
-            return True
-        return False  # no change done
+            return False  # no change done
+            # self._create_sdk_res(conn, sdk_params)
+            # return True
```

Comment:
For a resource without `name`, this method will never do antything. That's not correct. I would say, rather than editing the Resource implementation here, just fully override the `create_or_update` method in the SecurityGroupRule class. 
If you want insipration you can look at RouterInterface:

https://github.com/os-migrate/os-migrate/blob/684ff405ce7b8e44c51beb24d0b0baef154e56f7/os_migrate/plugins/module_utils/router_interface.py#L54

That resource also doesn't use a `name` as a key to look up existing one, rather it uses a combination of network_name + subnet_name + IP address:

https://github.com/os-migrate/os-migrate/blob/684ff405ce7b8e44c51beb24d0b0baef154e56f7/os_migrate/plugins/module_utils/router_interface.py#L87

RouterInterface is very similar to SecurityGroupRule, it also gets exported in bulk per one router, but it gets imported per-interface, like the export_security_group_rules and import_security_group_rule. Just the creation logic is a bit more complex in RouterInterface because it's done in 2 steps (create port, attach to router). However, structure-wise it might serve as a good pattern perhaps.

Reply:
done :) I forgot to add my idenpotency check. And this was the place to add it.